### ISSUE TITLE
custom Auto-type string not working

### ISSUE THREAD

**Issue by harvald** [keeweb/keeweb#1858]
Kee-web version: 1.18.6

I cannot set custom keystroke for auto-type function to work properly. It uses wrong keystrokes as if option would be not set.

For entry i have set options:

> Auto-type: "Enable auto-type for this entry" - checked
> Keystrokes: "{USERNAME}{TAB}{ENTER}{DELAY 150}{PASSWORD}{DELAY 50}{TAB}{SPACE}{TAB}"

In console:

> 2021-07-13T08:29:08.380Z [auto-type] Auto type event null
index.html:1 2021-07-13T08:29:08.381Z [auto-type] Getting window info
index.html:1 2021-07-13T08:29:08.400Z [auto-type] Window info undefined Logowanie - BitBay Auth — Mozilla Firefox null
index.html:1 2021-07-13T08:29:08.401Z [auto-type] processing auto-type event
index.html:1 2021-07-13T08:29:08.404Z [auto-type] Hide window
index.html:1 2021-07-13T08:29:08.406Z [auto-type] Start {USERNAME}{ENTER}{DELAY 2000}{PASSWORD}{ENTER}
index.html:1 2021-07-13T08:29:08.406Z [auto-type] Parsed [op:USERNAME,op:ENTER,op:DELAY,op:PASSWORD,op:ENTER]
index.html:1 2021-07-13T08:29:08.407Z [auto-type] Resolved [**********************,key:enter,cmd:wait,[*,*,*,*,*,*,*,*,*,*,*,*],key:enter]
index.html:1 2021-07-13T08:29:10.637Z [auto-type] Complete 2231ms

**Comment by antelle** [comment]
Hi! What do you mean as "custom keystroke"?

**Comment by harvald** [comment]
Keystroke other than default which is "{USERNAME}{TAB}{PASSWORD}{ENTER}"

**Comment by antelle** [comment]
So, what do you see typed if you enter your sequence?

**Comment by harvald** [comment]
You misunderstood me. In program my sequence is shown like it should, but when auto-typing function is used, then instead of mine, the other one is used like I showed in log.

**Comment by antelle** [comment]
Yes I understood that it displays it right but types incorrectly. So, it's just typing username/password, right?

**Comment by antelle** [comment]
I can't reproduce it, does it happen on other apps, e.g. a text editor?

**Comment by harvald** [comment]
Today, (so after computer restart) it seems fine (in browser and in text editor). So my suspicion is that after creating new entry, setting keystroke, something was not set correctly and stay in ram, even after restarting program.

**Comment by antelle** [comment]
If this repeats, can you please describe your actions that lead to this behavior?

**Comment by harvald** [comment]
Once again I've created new entry. (my kdbx is served over webdav)
1. Create new entry.
2. Fill user and password fields.
3. Try use auto-type on the website. 
4. (logs should show used keystroke {USERNAME}{ENTER}{DELAY 2000}{PASSWORD}{ENTER} )
5. Enter keystroke {USERNAME}{ENTER}{PASSWORD}{ENTER}
6. Try use auto-type on the website.
7. (logs should still show used keystroke {USERNAME}{ENTER}{DELAY 2000}{PASSWORD}{ENTER} )
8. Enter keystroke {USERNAME}{ENTER}{DELAY 1000}{PASSWORD}{TAB}{SPACE}{TAB}
9. Try use auto-type on the website.
10. (logs should still show used keystroke {USERNAME}{ENTER}{DELAY 2000}{PASSWORD}{ENTER} )
11. Sync changes in kdbx file with server.
12. Try use auto-type on the website.
13. (logs should still show used keystroke {USERNAME}{ENTER}{DELAY 2000}{PASSWORD}{ENTER} )

**Comment by antelle** [comment]
How did you trigger auto-type, from keeweb of from the website?

**Comment by harvald** [comment]
By key combination. I set it to "alt+shift+Q".

**Comment by antelle** [comment]
What if you trigger auto-type from KeeWeb? I was thinking, it's most likely because you don't "unfocus" the field, and it's not saved yet when you trigger auto-typing.

**Comment by harvald** [comment]
I've tried manually locking file, closing KeeWeb and synchronizing file with server (via webdav). No difference.

**Comment by antelle** [comment]
Does it repeat on the demo file?

**Comment by harvald** [comment]
On demo file it is working fine.

**Comment by antelle** [comment]
do you have any auto-type settings for folders in your file?

**Comment by harvald** [comment]
No.

**Comment by antelle** [comment]
Interesting. What about a new file?

**Comment by harvald** [comment]
New local file, new file through webdav, and new file imported from keepass - all working okay.
This suggest that my file is corrupted in some way, but I haven't notice any other issues.

**Comment by antelle** [comment]
Not good, in this case I'll need some effort from you to reproduce it. Maybe you can try moving the entry to the root, different folders, check folder settings, etc...
If you uncheck enable auto-type, you won't be find this entry in the auto-type selection box, but you can still do auto-typing from it.

**Comment by harvald** [comment]
LOL. If entry title is "bitbay.net" then the issue appear. If I change it to i.e. "bit", then ask me to choose entry and it works okay. Change it back and will still occurs.
Website field is empty. 
Website address: `https://auth.bitbay.net/login-password`
Web browser (Firefox) windows title: `Logowanie - BitBay Auth — Mozilla Firefox`

**Comment by antelle** [comment]
What a find! Thank you, I'll look into it
